### PR TITLE
Post Processing API renaming to be consistent

### PR DESCRIPTION
**Pull request by romanlutz**
- constraints instead of criteria
- name-only argument sensitive_features
- renaming attributes to sensitive_feature(s) wherever possible
- model -> predictor
- grid_size instead of gridsize
- estimator instead of unconstrained_estimator

### CODE REVIEW

File: fairlearn/post_processing/threshold_optimizer.py [hunk 1]
```diff
@@ -37,26 +37,26 @@
 UNEXPECTED_DATA_TYPE_ERROR_MESSAGE = "Unexpected data type {} encountered."
 
 
-_SUPPORTED_PARITY_CRITERIA = [DEMOGRAPHIC_PARITY, EQUALIZED_ODDS]
+_SUPPORTED_constraints = [DEMOGRAPHIC_PARITY, EQUALIZED_ODDS]
```
Comment (riedgar-ms) [inline fairlearn/post_processing/threshold_optimizer.py]: Casing

Comment (romanlutz) [reply fairlearn/post_processing/threshold_optimizer.py]: oh no I missed one!

File: fairlearn/post_processing/post_processing.py [hunk 2]
```diff
@@ -1,31 +1,31 @@
...
+                 constraints=None):
+        if unconstrained_predictor and estimator:
             raise ValueError(EITHER_MODEL_OR_ESTIMATOR_ERROR_MESSAGE)
-        elif unconstrained_model:
-            self._unconstrained_model = unconstrained_model
-            self._unconstrained_estimator = None
+        elif unconstrained_predictor:
+            self._unconstrained_predictor = unconstrained_predictor
+            self._estimator = None
             self._validate_model()
```
Comment (MiroDudik) [inline fairlearn/post_processing/post_processing.py]: change to _validate_predictor?

File: fairlearn/post_processing/threshold_optimizer.py [hunk 3]
```diff
@@ -37,26 +37,26 @@
...
                  random_state=None):
         """ Creates the post processing object.
 
-        :param unconstrained_model: the trained model whose output will be post processed
-        :type unconstrained_model: a trained model
-        :param unconstrained_estimator: an untrained estimator that will be trained, and
+        :param unconstrained_predictor: the trained model whose output will be post processed
+        :type unconstrained_predictor: a trained model
+        :param estimator: an untrained estimator that will be trained, and
             subsequently its output will be post processed
```
Comment (MiroDudik) [inline fairlearn/post_processing/threshold_optimizer.py]: trained model -> trained predictor

File: fairlearn/post_processing/post_processing.py (hunk 2 above)
Comment (romanlutz) [reply fairlearn/post_processing/post_processing.py]: Right, thanks!

File: fairlearn/post_processing/threshold_optimizer.py (hunk 3 above)
Comment (romanlutz) [reply fairlearn/post_processing/threshold_optimizer.py]: Thanks!

**Pull request merged**